### PR TITLE
Escapes binary path of wkhtmltopdf on windows

### LINKED ISSUE #112: wkthmltopdf engine not working in 3.0?
I'm following the instructions from https://github.com/FriendsOfCake/CakePdf/tree/3.0 to generate a pdf, the 2nd way described there to use it as email attachment.

I'm using the wk engine but it looks like the documentation isn't complete, there is a bug or i do something wrong. I have a model method that pretty much is the same as the example from the documentation.

When I run my test I'm getting:

```
Cake\Core\Exception\Exception: WKHTMLTOPDF didn't return any data
```

Debugging the command it generates has shown this:

```
C:\Program Files\wkhtmltopdf\bin\wkhtmltopdf.exe --quiet --print-media-type --orientation "portrait" --page-size "A4" --encoding "UTF-8" - -
```

It looks like the input file/data is missing?

### PR DESCRIPTION
**Pull request by rafaelios**
Resolves #112

### CODE REVIEW

File: src/Pdf/Engine/WkHtmlToPdfEngine.php [hunk 1]
```diff
@@ -16,13 +16,22 @@ class WkHtmlToPdfEngine extends AbstractPdfEngine
     protected $_binary = '/usr/bin/wkhtmltopdf';
 
     /**
+     * Flag to indicate if the environment is windows
+     *
+     * @access protected
```
Comment (dereuromark) [inline src/Pdf/Engine/WkHtmlToPdfEngine.php]: no access please :)

File: src/Pdf/Engine/WkHtmlToPdfEngine.php [hunk 2]
```diff
@@ -16,13 +16,22 @@ class WkHtmlToPdfEngine extends AbstractPdfEngine
...
+     * @var boolean
```
Comment (dereuromark) [inline src/Pdf/Engine/WkHtmlToPdfEngine.php]: bool

File: src/Pdf/Engine/WkHtmlToPdfEngine.php [hunk 3]
```diff
@@ -16,13 +16,22 @@ class WkHtmlToPdfEngine extends AbstractPdfEngine
...
+
+    /**
      * Constructor
      *
      * @param CakePdf $Pdf CakePdf instance
      */
     public function __construct(CakePdf $Pdf)
     {
         parent::__construct($Pdf);
+        $this->_windowsEnvironment = strtoupper(substr(PHP_OS, 0, 3) === 'WIN');
```
Comment (dereuromark) [inline src/Pdf/Engine/WkHtmlToPdfEngine.php]: Sure that works? Looks to be the second ) is place wrongly

Comment (rafaelios) [inline src/Pdf/Engine/WkHtmlToPdfEngine.php]: Oh, my mistake. I fixed it.

**Comment by rafaelios** [comment]
Could anyone review this pull request?

**Comment by ADmad** [comment]
@dereuromark Can you try out this change and confirm it's good to go?

File: src/Pdf/Engine/WkHtmlToPdfEngine.php [hunk 4]
```diff
@@ -10,19 +10,26 @@ class WkHtmlToPdfEngine extends AbstractPdfEngine
...
      * @var string
      */
     protected $_binary = '/usr/bin/wkhtmltopdf';
 
     /**
+     * Flag to indicate if the environment is windows
+     *
+     * @var bool
+     */
+    private $_windowsEnvironment;
```
Comment (dereuromark) [inline src/Pdf/Engine/WkHtmlToPdfEngine.php]: should be protected

**Comment by dereuromark** [comment]
Dont have time, but looks legit to me.

File: src/Pdf/Engine/WkHtmlToPdfEngine.php (hunk 4 above)
Comment (rafaelios) [inline src/Pdf/Engine/WkHtmlToPdfEngine.php]: @dereuromark Sure, I made the change, but I'd like to understand why.

**Pull request merged**